jobbergate-api/jobbergate_api/tests/apps/job_submissions/test_routers.py

```diff
@@ -108,6 +108,90 @@ async def test_create_job_submission__with_client_id_in_token(
+    request is made, the correct status code (201) is returned. We also show that the ``client_id``
+    is pulled from the token and the created job_submission is connected to that client id.
+
+    This test is the same as the previous one, but it does not include the ``execution_parameters``
+    in the payload, since they are optional.
+    """
+    inserted_application_id = await database.execute(
+        query=applications_table.insert(),
+        values=fill_application_data(),
+    )
+    inserted_job_script_id = await database.execute(
+        query=job_scripts_table.insert(),
+        values=fill_job_script_data(application_id=inserted_application_id),
+    )
+
+    inject_security_header(
+        "owner1@org.com",
+        Permissions.JOB_SUBMISSIONS_EDIT,
+        client_id="dummy-cluster-client",
+    )
+    create_data = fill_job_submission_data(
+        job_script_id=inserted_job_script_id,
+        job_submission_name="sub1",
+        job_submission_owner_email="owner1@org.com",
+    )
+
+    # Removed defaults to make sure these are correctly set by other mechanisms
+    create_data.pop("status", None)
+    create_data.pop("client_id", None)
+
+    with mock.patch(
+        "jobbergate_api.apps.job_submissions.routers.get_job_properties_from_job_script"
+    ) as mocked:
+        mocked.return_value = JobProperties()
+        with time_frame() as window:
+            response = await client.post("/jobbergate/job-submissions/", json=create_data)
+        mocked.assert_called_once_with(inserted_job_script_id)
+
+    assert response.status_code == status.HTTP_201_CREATED
+
+    count = await database.fetch_all("SELECT COUNT(*) FROM job_submissions")
+    assert count[0][0] == 1
+
+    id_rows = await database.fetch_all("SELECT id FROM job_submissions")
+    assert len(id_rows) == 1
+
+    job_submission = JobSubmissionResponse(**response.json())
+
+    # Check that the response correspond to the entry in the database
+    job_submission_raw_data = await database.fetch_one(
+        query=job_submissions_table.select().where(job_submissions_table.c.id == id_rows[0][0])
+    )
+    assert job_submission_raw_data is not None
+    assert job_submission == JobSubmissionResponse(**job_submission_raw_data)  # type: ignore
+
+    # Check that each field is correctly set
+    assert job_submission.id == job_submission_raw_data.get("id")
+    assert job_submission.job_submission_name == "sub1"
+    assert job_submission.job_submission_owner_email == "owner1@org.com"
+    assert job_submission.job_submission_description is None
+    assert job_submission.job_script_id == inserted_job_script_id
+    assert job_submission.execution_directory is None
+    assert job_submission.client_id == "dummy-cluster-client"
+    assert job_submission.status == JobSubmissionStatus.CREATED
+    assert job_submission.created_at in window
+    assert job_submission.updated_at in window
```

Comment:
so you're checking if the job submission data returned in the response is the one that is in the database. Despite it kind of making sense, it is possible the test is not testing the reality. Check the following piece of code:

```python
@app.post("/whatever")
async def whatever(body: Model):
    # build a row based on nothing and completely ignore the input model
    body = Model(jobbergate="is cool", testing="is not cool")

    # logic to insert the body model in the database
```

In this code I showed you that your test would succeed but the data in the database isn't what you really want.

So, to fix the test is simple: **you need to check the database data with the `create_data` variable**.

Does it make sense to you?

Reply:
> On the other hand, on lines 181-190, we make sure that each field has the expected value based on the test setup, which confirms that the data on both response and database is what we want

In line 168 you just select a random job submisson from the database. You **expect** it to be the one created by the endpoint but you cannot guarantee it is the same unless you check against the `create_data` variable. The `create_data` variable defined in line 145 is only used as payload for the request, by my snippet precisely showed you that you can completely ignore the payload, create an entry in the database and then make the test a fool test.

Reply:
Thanks for your suggestion, it was addressed on [ba92d0b](https://github.com/omnivector-solutions/jobbergate/pull/215/commits/ba92d0b54fe5114cc0a5ecc54e2db76f90c01669)